src/transaction_builder.js

```diff
@@ -459,6 +460,48 @@ export class TransactionBuilder {
     return this;
   }
 
+  /**
+   * Sets the ledger access footprint of the Soroban transaction to match the
+   * given parameters, creating a {@link xdr.SorobanTransactionData} structure
+   * if it doesn't exist.
+   *
+   * This is a convenience method for building bump / restore operations without
+   * need to build the entire data structure from scratch and use
+   * {@link TransactionBuilder.setSorobanData}.
+   *
+   * @param {xdr.LedgerKey[] | null} readOnly   the set of ledger keys to set in
+   *    the read-only portion of the transaction's `sorobanData`. if null is
+   *    passed, the field is left untouched (so if you want to clear it, pass an
+   *    empty array)
+   * @param {xdr.LedgerKey[] | null} readWrite  the set of ledger keys to set in
+   *    the read-write portion of the transaction's `sorobanData`. if null is
+   *    passed, the field is left untouched (so if you want to clear it, pass an
+   *    empty array)
+   *
+   * @returns {TransactionBuilder}
+   *
+   * @warning If the `sorobanData` has NOT been set (e.g. via
```

Comment:
not suggesting change, but seeing two different methods on builder that overwrite same state is not ideal, another option may be to go with builder pattern like   `SorobanTransactionDataBuilder` in resources.js, and then no changes in transaction_builder, external client code looks something like:
```
// tx data builder creates from nothing or an existing xdr instance as initial state
SorobanTransactionDataBuilder dataBuilder = new SorobanTransactionDataBuilder(null | xdr.SorobanTransactionData);
dataBuilder.setReadOnlyFootprint(roFootPrint);
dataBuilder.setReadWriteFootprint(rwFootPrint);
sorobanTxData = dataBuilder.build()

TransactionBuilder tb = new TransactionBuilder();
...
tb.setSorobanData(sorobanTxData)
...
// something triggers change on footprints, want to rebuild the tx
dataBuilder = new SorobanTransactionDataBuilder(tb.sorobanData);
dataBuilder.setReadOnlyFootprint(newRoFootPrint);
tb.setSorobanData(dataBuilder.build())
newerTx = tb.build()
```

Reply:
>two different methods on builder that overwrite same state is not ideal 

There is precedent with `.setTimebounds()` and `.setTimeout()`, heh.

Your design is quite elegant, though! I like it. My only concern is that there are probably not very many other things you'd want to set with this pattern besides the footprint (hence why the `.setFootprint()` subset isn't so bad), since e.g. the fees nor extension should almost certainly never be set directly. I like it, though! So it's done in 46ab05be9d225d5c3b3e8cd815f667362f08b52b.